apps/gdalalg_raster_contour.cpp

```diff
@@ -0,0 +1,221 @@
+/******************************************************************************
+ *
+ * Project:  GDAL
+ * Purpose:  gdal "raster contour" subcommand
+ * Author:   Alessandro Pasotti <elpaso at itopen dot it>
+ *
+ ******************************************************************************
+ * Copyright (c) 2025, Alessandro Pasotti <elpaso at itopen dot it>
+ *
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+
+#include "gdalalg_raster_contour.h"
+
+#include "cpl_conv.h"
+#include "gdal_priv.h"
+#include "gdal_utils.h"
+#include "gdal_alg.h"
+#include "gdal_utils_priv.h"
+
+//! @cond Doxygen_Suppress
+
+#ifndef _
+#define _(x) (x)
+#endif
+
+/************************************************************************/
+/*          GDALRasterContourAlgorithm::GDALRasterContourAlgorithm()    */
+/************************************************************************/
+
+GDALRasterContourAlgorithm::GDALRasterContourAlgorithm(
+    bool openForMixedRasterVector)
```

Comment:
you don't need that flag for contour (this is only useful for things like "gdal info" that tries to forward either to "gdal raster info" or "gdal vector info")
```suggestion
)
```